app/components/post_add_channel_member/index.js

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {Posts} from 'mattermost-redux/constants';
+import {PostTypes} from 'mattermost-redux/action_types';
+
+import {addChannelMember} from 'mattermost-redux/actions/channels';
+import {removePost} from 'mattermost-redux/actions/posts';
+
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+import {getChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+
+import {generateId} from 'app/utils/file';
+
+import PostAddChannelMember from './post_add_channel_member';
+
+function mapStateToProps(state, ownProps) {
+    const post = getPost(state, ownProps.postId) || {};
+    let channelType = '';
+    if (post && post.channel_id) {
+        const channel = getChannel(state, post.channel_id);
+        if (channel && channel.type) {
+            channelType = channel.type;
+        }
+    }
+
+    return {
+        channelType,
+        currentUser: getCurrentUser(state),
+        post,
+    };
+}
+
+function sendEphemeralPost(user, addedUsername, message, channelId, postRootId) {
```

Comment:
Can you move this to be an action under `app/actions/views` ?

Reply:
Sure. I should have add test to `sendAddToChannelEphemeralPost` but I'm having issue with the mock store. Let me do that separately (next time) if that's fine with you.